cmd/macpassd/dbConnection.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type MacRegistration struct {
+	mac  string
+	id   int
+	user string
+}
+
+func getActive(con *sql.DB) []MacRegistration {
+	selectOutdated := `select id,User,MAC from Log where ? > maxTime and isDown = true`
+	row, err := con.Query(selectOutdated, time.Now())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var macs []MacRegistration
+	var user string
+	var mac string
+	var id int
+	for row.Next() {
+		row.Scan(&id, &user, &mac)
+		macs = append(macs, MacRegistration{mac, 0, user})
+	}
+	return macs
+}
+func setOutdated(con *sql.DB) []MacRegistration {
+	selectOutdated := `select id,User,MAC from Log where ? < maxTime and isDown = false`
+	row, err := con.Query(selectOutdated, time.Now())
+	if err != nil {
+		log.Fatal(err)
+	}
+	var macs []MacRegistration
+	var user string
+	var mac string
+	var id int
+	for row.Next() {
+		row.Scan(&id, &user, &mac)
+		macs = append(macs, MacRegistration{mac, 0, user})
+	}
```

Comment:
```suggestion
	var macs []MacRegistration
	var tmp MacRegistration
	for row.Next() {
		row.Scan(&tmp.id, &tmp.user, &tmp.mac)
		macs = append(macs, tmp)
	}
```

Direi che è leggermente meglio così